### PR TITLE
1 get incoming transactions

### PR DESCRIPTION
**Pull request by lalvarezguillen**
Created `JSONRPCWallet.get_incoming_transactions` method, and created unittests for it

### CODE REVIEW

File: monero/backends/jsonrpc.py [hunk 1]
```diff
@@ -183,6 +184,47 @@ def balances(self, account=0):
...
+    ) -> List[IncomingPayment]:
+        """
+        Calls the wallet RPC to get incoming transactions.
+
+        Args:
+            account_index: The index of the account we want to query.
+            unconfirmed: Whether to include unconfirmed transactions (the ones
+            in the pool).
+            min_height: The height of the block where we want to start scanning
+            from.
```
Comment (dethos) [inline monero/backends/jsonrpc.py:246]: Docstring is missing information about `subaddr_indices`.

**Review by dethos — COMMENTED** [review]
Just one small observation. The rest looks good to me.

File: monero/backends/jsonrpc.py (hunk 1 above)
Comment (lalvarezguillen) [reply monero/backends/jsonrpc.py:246]: Good catch. It's updated now

**Comment by lalvarezguillen** [comment]
I decided to include in this PR the `get_unspent_inputs` method that  [#5](https://github.com/Domol/monero-python/issues/5) had defined

**Review by dethos — APPROVED** [review]
Looks good to me. However the naming `unspent inputs` doesn't sound that great and perhaps a little confusing, I understand what it means but the RPC does not use it and I never saw that expression anywhere.

**Comment by lalvarezguillen** [comment]
> Looks good to me. However the naming `unspent inputs` doesn't sound that great and perhaps a little confusing, I understand what it means but the RPC does not use it and I never saw that expression anywhere.

Lets run this by Jeremie before merging. He uses the term `input` often. I don't think `Transfer` or `Transaction` fit

File: monero/backends/jsonrpc.py [hunk 2]
```diff
@@ -183,6 +186,91 @@ def balances(self, account=0):
...
+            account_index: The index of the account whose Unspent Inputs we
+            want to fetch.
+            subaddr_indices: If it's provided, this function only fetches the
+            Unspent Inputs addressed to the Subaddress whose indices are
+            included in this list.
+            verbose: Whether to include key_image in the RPC response.
+        """
+        params = {
+            "transfer_type": "available",
+            'account_index': account_index,
```
Comment (domol) [inline monero/backends/jsonrpc.py:209]: I tried to find what is common practice for strings `'` or `"`. Did not find anything so you choose but try to be consistent in one dict ;)

File: monero/backends/jsonrpc.py [hunk 3]
```diff
@@ -183,6 +186,91 @@ def balances(self, account=0):
...
+            in the pool).
+            min_height: The height of the block where we want to start scanning
+            from.
+            subaddr_indices: If this is included, this function only fetches
+            the the Incoming Transactions addressed to the Subaddresses whose
+            indices are included in this list
+        """
+        params: Dict[str, Any] = {
+            'account_index': account_index,
+            "in": True,
```
Comment (domol) [inline monero/backends/jsonrpc.py]: same about strings.

File: monero/backends/jsonrpc.py [hunk 4]
```diff
@@ -183,6 +186,91 @@ def balances(self, account=0):
...
+            params['subaddr_indices'] = subaddr_indices
+
+        resp = self.raw_request("get_transfers", params)
+        transfers = resp.get("in", [])
+        if unconfirmed:
+            transfers.extend(resp.get("pool", []))
+
+        return list(map(self._inpayment, transfers))
+
+    # TODO: consider removing/deprecating
```
Comment (domol) [inline monero/backends/jsonrpc.py:274]: Why don't you create an issue describing why it should be deprecated?

**Review by domol — CHANGES_REQUESTED** [review]

**Comment by domol** [comment]
Do you guys think we should squash commits before merging?

File: monero/backends/jsonrpc.py (hunk 4 above)
Comment (lalvarezguillen) [reply monero/backends/jsonrpc.py:274]: Good point. Created: https://github.com/Domol/monero-python/issues/10

File: monero/backends/jsonrpc.py (hunk 3 above)
Comment (lalvarezguillen) [reply monero/backends/jsonrpc.py]: Fixed

File: monero/backends/jsonrpc.py (hunk 2 above)
Comment (lalvarezguillen) [reply monero/backends/jsonrpc.py:209]: Fixed. Standardised to `'`

**Comment by lalvarezguillen** [comment]
> Do you guys think we should squash commits before merging?

We probably should.

**Review by domol — APPROVED** [review]

**Pull request merged**